ably-ios/ARTEventEmitter+Private.h

```diff
@@ -0,0 +1,30 @@
+//
+//  ARTEventEmitter+Private.h
+//  ably
+//
+//  Created by Toni Cárdenas on 29/1/16.
+//  Copyright © 2016 Ably. All rights reserved.
+//
+
+#include "ARTEventEmitter.h"
+#include "CompatibilityMacros.h"
+
+ART_ASSUME_NONNULL_BEGIN
+
+@interface __GENERIC(ARTEventEmitterEntry, ItemType) : NSObject
+
+@property (readwrite, strong, nonatomic) __GENERIC(ARTEventListener, ItemType) *listener;
+@property (readwrite, nonatomic) BOOL once;
+
+- (instancetype)initWithListener:(__GENERIC(ARTEventListener, ItemType) *)listener once:(BOOL)once;
+
+@end
+
+@interface __GENERIC(ARTEventEmitter, EventType, ItemType) ()
+
+@property (readwrite, nonatomic) __GENERIC(NSMutableDictionary, EventType, __GENERIC(NSMutableArray, __GENERIC(ARTEventEmitterEntry, ItemType) *) *) *listeners;
+@property (readwrite, nonatomic) __GENERIC(NSMutableArray, __GENERIC(ARTEventEmitterEntry, ItemType) *) *totalListeners;
```

Comment:
Not mad about the name `totalListeners` as that makes it sound like it's a list compiled from all the listeners, whereas this this Array contains a list of listeners for any event.  Could it not be called something like `anyListeners` or `allEventListeners`?